internal/storage/bucket_handle.go

```diff
@@ -237,6 +237,44 @@ func (bh *bucketHandle) CreateObject(ctx context.Context, req *gcs.CreateObjectR
 	o = storageutil.ObjectAttrsToBucketObject(attrs)
 	return
 }
+func (bh *bucketHandle) CreateObjectChunkWriter(ctx context.Context, req *gcs.CreateObjectRequest, chunkSize int, callBack func(bytesUploadedSoFar int64)) (gcs.Writer, error) {
+	// For phase 1 of buffered writes, we are doing chunk uploads only for new
+	// file uploads.
+	preconditions := storage.Conditions{}
```

Comment:
why do you want to restrict for not exist scenario. why can't we just use the generation from the request

Reply:
I was restricting to phase 1 of the project but makes sense to do this in one go as well. I'll update this.